src/pages/workspace/distanceRates/PolicyDistanceRatesSettingsPage.tsx

```diff
@@ -65,10 +65,7 @@ function PolicyDistanceRatesSettingsPage({route}: PolicyDistanceRatesSettingsPag
             return;
         }
 
-        Category.setPolicyDistanceRatesDefaultCategory(policyID, customUnit, {
-            ...customUnit,
-            defaultCategory: defaultCategory === category.searchText ? '' : category.searchText,
-        });
+        Category.setPolicyCustomUnitDefaultCategory(policyID, customUnitID, customUnit.defaultCategory, defaultCategory === category.searchText ? '' : category.searchText);
```

Comment:
```suggestion
        if (category.searchText === defaultCategory) {
            return;
        }
        Category.setPolicyCustomUnitDefaultCategory(policyID, customUnitID, customUnit.defaultCategory, defaultCategory === category.searchText ? '' : category.searchText);
```

Same in `setNewCategory` of `WorkspacePerDiemSettingsPage` based on [this](https://github.com/Expensify/App/pull/52269/files#r1840310251).

@shubham1206agra